src/plugins/kubeconfig/index.ts

```diff
@@ -0,0 +1,138 @@
+**/
+import { KubeconfigCommandArgs } from "../../commands/kubeconfig";
+import { waitForProvisioning } from "../../commands/shared";
+import { request } from "../../commands/shared/request";
+import { doc } from "../../drivers/firestore";
+import { print2 } from "../../drivers/stdio";
+import { Authn } from "../../types/identity";
+import { Request } from "../../types/request";
+import { assertNever } from "../../util";
+import { assumeRoleWithIdc } from "../aws/idc";
+import { AwsCredentials } from "../aws/types";
+import { assumeRoleWithOktaSaml } from "../okta/aws";
+import {
+  EksClusterConfig,
+  K8sConfig,
+  K8sGenerated,
+  K8sPermissionSpec,
+} from "./types";
+import { getDoc } from "firebase/firestore";
+import { pick } from "lodash";
+import yargs from "yargs";
+
+export const getAndValidateK8sIntegration = async (
+  authn: Authn,
+  clusterId: string
+): Promise<EksClusterConfig> => {
+  const configDoc = await getDoc<K8sConfig, object>(
+    doc(`o/${authn.identity.org.tenantId}/integrations/k8s`)
+  );
+
+  const clusterConfig = configDoc
+    .data()
+    ?.workflows.items.find(
+      (c) => c.clusterId === clusterId && c.state === "installed"
+    );
+
+  if (!clusterConfig) {
+    throw `Cluster with ID ${clusterId} not found`;
+  }
+
+  const { awsAccountId, awsClusterArn } = clusterConfig;
+
+  if (!awsAccountId || !awsClusterArn) {
+    throw (
+      `This command currently only supports AWS EKS clusters, and ${clusterId} is not configured as one.\n` +
+      "You can request access to the cluster using the `p0 request k8s` command."
+    );
+  }
+
+  return {
+    ...clusterConfig,
+    awsAccountId,
+    awsClusterArn,
+  };
+};
+
+export const requestAccessToCluster = async (
+  authn: Authn,
+  args: yargs.ArgumentsCamelCase<KubeconfigCommandArgs>,
+  clusterId: string,
+  role: string
+): Promise<Request<K8sPermissionSpec>> => {
+  const response = await request("request")(
+    {
+      ...pick(args, "$0", "_"),
+      arguments: [
+        "k8s",
+        "resource",
+        "--cluster",
+        clusterId,
+        "--role",
+        role,
+        ...(args.resource ? ["--locator", args.resource] : []),
+        ...(args.reason ? ["--reason", args.reason] : []),
+        ...(args.requestedDuration
+          ? ["--requested-duration", args.requestedDuration]
+          : []),
+      ],
+      wait: true,
+    },
+    authn,
+    { message: "approval-required" }
+  );
+
+  if (!response) {
+    throw "Did not receive access ID from server";
+  }
+  const { id, isPreexisting } = response;
+  if (!isPreexisting) {
+    print2(
+      "Waiting for access to be provisioned. This may take up to a minute."
+    );
+  }
+
+  return await waitForProvisioning<K8sPermissionSpec>(authn, id);
+};
+
+export const profileName = (eksCluterName: string): string =>
+  `p0cli-managed-eks-${eksCluterName}`;
+
+export const awsCloudAuth = async (
```

Comment:
Since this is AWS specific it should live in `common/aws` somewhere.

Reply:
Technically yes, but this feels to me as more closely tied to Kubernetes as it is expecting arguments in a form that probably doesn't generalize easily without further work (i.e., I think moving this into `common/aws` might be an over-optimization at this time). Note that `assumeRoleWithIdc()` and `assumeRoleWithOktaSaml()` themselves do live within `/src/plugins/aws` and `/src/plugins/okta`, respectively.